app/src/main/java/com/tananaev/logcat/LineAdapter.java

```diff
@@ -30,7 +43,11 @@
     private List<Line> linesAll = new LinkedList<>();
     private List<Line> linesFiltered = new LinkedList<>();
 
-    private String keyword;
+    private String tag = "";
+    private String keyword = "";
+    private String lowerKeyword = "";
+    private String searchWord = "";
+    private String lowerSearchWord = "";
```

Comment:
lowerKeyword , lowerSearchword is used by cached value when viewHolder is updated. 
But is not important. I fixed